### PR TITLE
Add missing XML doc elements for methods in `RunspaceFactory`

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Related: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/9306
Add missing XML doc elements for methods in `RunspaceFactory`.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/System.Management.Automation/engine/hostifaces/ConnectionFactory.cs [hunk 1]
```diff
@@ -476,39 +476,43 @@ public static RunspacePool CreateRunspacePool(int minRunspaces, int maxRunspaces
...
         ///   2. SerializationDepth
         ///   3. SpecificSerializationProperties
+        ///
         /// TypeTable has the following information used by deserializer:
         ///   1. TargetTypeForDeserialization
         ///   2. TypeConverter
         /// </param>
-        /// <param name="host"></param>
-        /// <param name="connectionInfo"></param>
         /// <returns></returns>
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/hostifaces/ConnectionFactory.cs:494]: Still empty?

**Review by iSazonov — APPROVED** [review]
nit: There are still many `<returns></returns>`.

File: src/System.Management.Automation/engine/hostifaces/ConnectionFactory.cs (hunk 1 above)
Comment (sdwheeler) [reply src/System.Management.Automation/engine/hostifaces/ConnectionFactory.cs:494]: Need to fill in the `<returns>` element

**Review by sdwheeler — CHANGES_REQUESTED** [review]
We need to fill in all empty XML elements in this file.

File: src/System.Management.Automation/engine/hostifaces/ConnectionFactory.cs (hunk 1 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/hostifaces/ConnectionFactory.cs]: Updated.

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@
 namespace System.Management.Automation.Runspaces
 {
     /// <summary>
+    /// This class represents a PowerShell process that is used for an out-of-process remote Runspace.
```
Comment (sdwheeler) [inline src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs:14]: There are still empty XML tags in this file.

**Review by sdwheeler — CHANGES_REQUESTED** [review]
There still empty XML tags in the PowerShellProcessInstance.cs file.

File: src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs (hunk 2 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs:14]: Updated the tags on public members.

**Review by sdwheeler — APPROVED** [review]

**Pull request merged**